### PR TITLE
frequencies: add as923-2 and as923-3

### PR DESCRIPTION
**Pull request by shawaj**
Add as923-2 and as923-3 frequency plans. The base plan remains unchanged, but we have added the ability to define a frequency offset which shifts the entire plan in frequency to accommodate these new regions. AS923-1 represents the original AS923 plan; AS923-2 shifts the plan down 1.8MHz to support Indonesia and Viet Nam; AS923-3 shifts the plan down 6.6MHz to support Philippines, Cuba as well as many European and EU-style regulatory domain countries. More details at https://lora-alliance.org/resource_hub/rp2-101-lorawan-regional-parameters-2/

Change-type: patch
Signed-off-by: Aaron Shaw <shawaj@gmail.com>

[rp_2-1.0.1.pdf](https://github.com/helium/miner/files/6043723/rp_2-1.0.1.pdf)

### CODE REVIEW

**Comment by shawaj** [comment]
Also will mean support for taiwan, thailand, laos, hong kong, cambodia and brunei and probably some others :-P

**Comment by Vagabond** [comment]
I think https://github.com/helium/miner/blob/master/priv/countries_reg_domains.csv needs to be updated as well to point at the specific sub region. @michaeldjeffrey is working on AS923 support in the router right now so I'm tagging him in here.

**Comment by shawaj** [comment]
Thanks @Vagabond totally missed that. My bad. Will fix and rebase

**Comment by Vagabond** [comment]
Can you grab the AS923 subregions for the CSV over from that other PR where I took everything but those?

Additionally we need to add packet forwarder config files for these regions like we have over here: https://github.com/helium/sx1302_hal/tree/helium/hotspot/packet_forwarder @jay or @lthiery can help with that

Finally we need to coordinate with the router team so we can decide if we need to add more regions to https://github.com/helium/proto/blob/9d604b5908ab6084676d3b6e4819792b0c11bc0e/src/blockchain_state_channel_v1.proto#L7 or if we want to treat them all the same (which would involve some simple logic to chop off the sub region for AS923).

**Comment by jmarcelino** [comment]
https://docs.helium.com/lorawan-on-helium/frequency-plans will also need updating

**Comment by Vagabond** [comment]
cc @michaeldjeffrey for what router wants to do about AS923 subregions

**Comment by lthiery** [comment]
@shawaj Where did you get these specific frequency plans? I started looking more into this as I noticed the AS2 did not line up with the AS2 plan [proposed by TTN](https://www.thethingsnetwork.org/docs/lorawan/frequency-plans.html#as923-925-as2).

Moreover, your proposed AS2 implements a 100 kHz spacing between the first two channels.
```
'AS923-2' => [921.5, 921.6, 921.8, 922.0, 922.2, 922.4, 922.6, 922.7, 922.8, 923.0],
```

The specification you linked says these two must be spaced by 200 kHz

image

Easy fix to set the first channel to 921.4 MHz but I'm still confused what the general philosophy is for these channel plans and why it disagrees with the one proposed by TTN.

File: config/sys.config [hunk 1]
```diff
@@ -100,6 +100,8 @@
                       'CN779' => [779.5, 779.7, 779.9],
                       'AU915' => [916.8, 917.0, 917.2, 917.4, 917.5, 917.6, 917.8, 918.0, 918.2],
                       'AS923' => [923.2, 923.4, 923.6, 923.8, 924.0, 924.2, 924.4, 924.5, 924.6, 924.8],
```
Comment (lthiery) [inline config/sys.config:102]:
```suggestion
                      'AS923-AS1' => [923.2, 923.4, 923.6, 923.8, 924.0, 924.2, 924.4, 924.5, 924.6, 924.8],
```

File: config/sys.config [hunk 2]
```diff
@@ -100,6 +100,8 @@
...
+                      'AS923-2' => [921.5, 921.6, 921.8, 922.0, 922.2, 922.4, 922.6, 922.7, 922.8, 923.0],
```
Comment (lthiery) [inline config/sys.config:103]:
```suggestion
                      'AS923-AS2' => [921.5, 921.6, 921.8, 922.0, 922.2, 922.4, 922.6, 922.7, 922.8, 923.0],
```

File: config/sys.config [hunk 3]
```diff
@@ -100,6 +100,8 @@
...
+                      'AS923-3' => [916.6, 916.8, 917.0, 917.2, 917.4, 917.6, 917.8, 917.9, 918.0, 918.2],
```
Comment (lthiery) [inline config/sys.config:104]:
```suggestion
                      'AS923-AS3' => [916.6, 916.8, 917.0, 917.2, 917.4, 917.6, 917.8, 917.9, 918.0, 918.2],
```

**Review by lthiery — CHANGES_REQUESTED** [review]
We discussed internally (@JayKickliter @michaeldjeffrey) and would like to go towards this naming convention.

File: config/sys.config (hunk 2 above)
Comment (lthiery) [reply config/sys.config:103]: > We discussed internally (@JayKickliter @michaeldjeffrey) and would like to go towards this naming convention.

My bad. Our agreed convention aligns with what you were doing. Only the 1st one needs the `-1` appended.

**Comment by lthiery** [comment]
> We discussed internally (@JayKickliter @michaeldjeffrey) and would like to go towards this naming convention.

My bad - I misunderstood our internal alignment. It actually converges with what you were doing, except we'd like to rename `AS923` to `AS923-1`

**Comment by shawaj** [comment]
@lthiery the link here is where it mentions the downshift...
https://lora-alliance.org/resource_hub/rp2-101-lorawan-regional-parameters-2/

Screenshot_20210329-055919_Chrome

So my understanding was that the standard plan was shifted down by those amounts. And this is also confirmed because in the doc it says in the table:
AS923-1 923-925 MHz
AS923-2 920-923 MHz

The 921.5 is a typo. It should have been 923.2 - 1.8 = 921.4

@Vagabond I'll bring in the changes from #663 into this no problem.

For the config files that's no issue either I think we may have them already somewhere in one of our repos

**Comment by shawaj** [comment]
On a side note @lthiery it looks like TTN "AS2" is actually the equivalent of LoRa alliance AS923-1.

I don't know why TTN have those the way they do but it seems like they've tried to maintain 923.2 and 923.4 as common channels on both their "as1" and "as2" plans. However this is wrong as per the LoRa doc as it quite clearly defined a negative offset even for those freqs

**Comment by lthiery** [comment]
@shawaj Thanks, that really clarifies things. 

If you have a time for a PR against [our sx1302_hal](https://github.com/helium/sx1302_hal), I'll close out [my attempt to iron these out](/pull/11). If you don't, I'll just fix that PR now that I finally understand what's going on.

**Comment by lthiery** [comment]
@shawaj Sorry one more question. How are the 9th and 10th channels determined in your map? I don't think the miner has any notion of FSK or the fat LoRa channel and I didn't see anything about them in the AS923 spec. Have I missed something again?

File: config/sys.config (hunk 2 above)
Comment (lthiery) [inline config/sys.config:103]:
```suggestion
                      'AS923-2' => [921.4, 921.6, 921.8, 922.0, 922.2, 922.4, 922.6, 922.7, 922.8, 923.0],
```

**Review by lthiery — CHANGES_REQUESTED** [review]

File: config/sys.config (hunk 1 above)
Comment (lthiery) [inline config/sys.config:102]:
```suggestion
                      'AS923-1' => [923.2, 923.4, 923.6, 923.8, 924.0, 924.2, 924.4, 924.6],
```

Reduce to 8 channels and rename to AS923-1

File: config/sys.config [hunk 4]
```diff
@@ -100,6 +100,8 @@
                       'CN779' => [779.5, 779.7, 779.9],
                       'AU915' => [916.8, 917.0, 917.2, 917.4, 917.5, 917.6, 917.8, 918.0, 918.2],
                       'AS923' => [923.2, 923.4, 923.6, 923.8, 924.0, 924.2, 924.4, 924.5, 924.6, 924.8],
+                      'AS923-2' => [921.4, 921.6, 921.8, 922.0, 922.2, 922.4, 922.6, 922.7, 922.8, 923.0],
```
Comment (lthiery) [inline config/sys.config:103]:
```suggestion
                      'AS923-2' => [921.4, 921.6, 921.8, 922.0, 922.2, 922.4, 922.6, 922.8],
```
Reduce to 8 channels

File: config/sys.config [hunk 5]
```diff
@@ -100,6 +100,8 @@
...
+                      'AS923-3' => [916.6, 916.8, 917.0, 917.2, 917.4, 917.6, 917.8, 917.9, 918.0, 918.2],
```
Comment (lthiery) [inline config/sys.config:104]:
```suggestion
                      'AS923-3' => [916.6, 916.8, 917.0, 917.2, 917.4, 917.6, 917.8, 918.0],
```
Reduce to 8 channels

**Review by lthiery — COMMENTED** [review]
I think we should reduce to 8 evenly spaced channels. I don't see a 500 kHz uplink datarate in the spec and no mention of FSK for these regions.

**Review by JayKickliter — CHANGES_REQUESTED** [review]
If we can get the current three suggestions merged I'll merge this PR.

**Comment by shawaj** [comment]
superseded by #772

**Comment by shawaj** [comment]
@JayKickliter have updated these changes in a new PR #772

**Comment by shawaj** [comment]
Have also updated the documentation in https://github.com/helium/docs/pull/345

**Pull request closed without merging**